### PR TITLE
Fix #57 - add `headers` parameter to methods create, update and delete

### LINKED ISSUE #57: add a headers arg to create_entity()
Hi,
Thank you for this nice wrapper.

**Describe the bug**
Sometimes I run into this error https://github.com/phpipam/phpipam/issues/3177
and as stated in this issue, I have to pass {"content-type: "application/x-www-form-urlencoded"} 
in the headers to solve it.
Currently create_entity() does not have a 'headers' argument, so I have to call _query() directly to pass the header to the request.

**Expected behavior**
It is a bite dirty to have to call an internal function directly, so I would like to
be able to pass an 'headers' arg to create_entity(), and also to update_entity() and delete_entity for consistency I guess.

**Versions:**
- python 3.10
- phpypam 1.0.2

### PR DESCRIPTION
**Pull request by positiveEV**
* Add `headers` parameter to methods `create`, `update` and `delete`
* Pass `headers` to internal method `_query`

### CODE REVIEW

**Comment by cmeissner** [comment]
@positiveEV thank you for submitting your first PR. To get it merged you need to fix some small things. 

1. Format the title sth like "Fix \<issue-id> - short description of issue"
2. Add a description what you have changed
3. Add a change log entry use `changelog` command (came from requirements-dev.txt)
4. Add a test case for the use case which is covered by your PR
    As you can't provide any steps to reproduce the issue the test case can only check the success but not the failure state
    
After that I think we can merge the PR as it seems not to interfere the existing test cases.

**Review by cmeissner — CHANGES_REQUESTED** [review]
LGTM but test case is still missing

**Comment by positiveEV** [comment]
Sorry, for not coming back at you earlier.
Regarding the test case if I write a ensure_address.py, and pass a headers arg to create_entity, will it be ok?

**Comment by cmeissner** [comment]
No matter. I had some time today to fix some points of the list. Two last things:

1. Can you please add headers parameter also to `get_entity` method?
2. Regarding the test case. As we can't reproduce the issue the test should test the documented feature to change the output from json to xml with `Content-Type: application/xml`. Here would be a ensure_address (CRUD test) a good idea. But instead of working with json responses work with xml.

File: phpypam/core/api.py [hunk 1]
```diff
@@ -153,20 +153,23 @@ def get_entity(self, controller, controller_path=None, params=None):
...
         _path = controller
         _controller_path = controller_path
         _params = params
+        _headers = headers
 
         if _controller_path:
             _path = '{}/{}'.format(_path, _controller_path)
 
-        return self._query(token=self._api_token, method=GET, path=_path, params=_params)
+        return self._query(token=self._api_token, method=GET, path=_path, params=_params, headers=None)
```
Comment (cmeissner) [inline phpypam/core/api.py:170]: This is an error, you need to pass `_headers` here.

**Review by cmeissner — CHANGES_REQUESTED** [review]
This does not work out. If you run `pi.get_entity(controller='subnets', controller_path='1', headers={'Content-Type': 'application/xml'})` (after applying the requested change in `get_entity` method) you run into an exception.

```python
pi.get_entity(controller='subnets', controller_path='1', headers={'Content-Type': 'application/xml'})
Traceback (most recent call last):
  File "/home/cme/.pyenv/versions/phpypam_3.9.11/lib/python3.9/site-packages/requests/models.py", line 971, in json
    return complexjson.loads(self.text, **kwargs)
  File "/home/cme/.pyenv/versions/3.9.11/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/home/cme/.pyenv/versions/3.9.11/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/cme/.pyenv/versions/3.9.11/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/cme/Development/github/username/phpypam/phpypam/core/api.py", line 170, in get_entity
    return self._query(token=self._api_token, method=GET, path=_path, params=_params, headers=_headers)
  File "/home/cme/Development/github/username/phpypam/phpypam/core/api.py", line 121, in _query
    result = resp.json()
  File "/home/cme/.pyenv/versions/phpypam_3.9.11/lib/python3.9/site-packages/requests/models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

This is related to [result = resp.json()](https://github.com/positiveEV/phpypam/blob/f7a8e3ffec260b2d6e4d12ecaab967d949a54b05/phpypam/core/api.py#L121) in `_query` method. You need to refactor this method, otherwise all other methods will break if you use `Content-Type: application/xml`.

**Comment by cmeissner** [comment]
This PR was closed because it has been stalled for 15 days with no activity.

**Pull request closed without merging**